components/camel-zookeeper/src/main/java/org/apache/camel/component/zookeeper/ZooKeeperProducer.java

```diff
@@ -64,6 +64,9 @@ public ZooKeeperProducer(ZooKeeperEndpoint endpoint) {
 
     public void process(Exchange exchange) throws Exception {
 
+        if (connection == null) {
```

Comment:
Could the change to ZooKeeperProducer.java be merged to v2.18.x?
(see also https://issues.apache.org/jira/browse/CAMEL-10961)